spynnaker/pyNN/abstract_spinnaker_common.py

```diff
@@ -347,6 +349,96 @@ def reset_number_of_neurons_per_core(self):
+
+        # if using extra monitor functionality, locate extra data items
+        receivers = list()
+        if using_extra_monitor_functionality:
+            receivers = self._locate_receivers_from_projections(
+                projection_to_attribute_map.keys(),
+                self.get_generated_output(
+                    "MemoryMCGatherVertexToEthernetConnectedChipMapping"),
+                self.get_generated_output(
+                    "MemoryExtraMonitorToChipMapping"))
+
+        # set up the router timeouts to stop packet loss
+        if using_extra_monitor_functionality:
+            for data_receiver, extra_monitor_cores in receivers:
+                data_receiver.set_cores_for_data_extraction(
+                    self._txrx, list(extra_monitor_cores), self._placements)
+
+        # acquire the data
+        for projection in projection_to_attribute_map:
+            for attribute in projection_to_attribute_map[projection]:
+                data = projection._get_synaptic_data(
+                    as_list=True, data_to_get=attribute,
+                    fixed_values=None, notify=None,
+                    handle_time_out_configuration=False)
+                mother_lode.set(projection, attribute, data)
+
+        # reset time outs for the receivers
+        if using_extra_monitor_functionality:
+            for data_receiver, extra_monitor_cores in receivers:
+                data_receiver.unset_cores_for_data_extraction(
+                    self._txrx, list(extra_monitor_cores), self._placements)
+
+        # return data items
+        return mother_lode
+
+    def _locate_receivers_from_projections(
+            self, projections, gatherers, extra_monitors_per_chip):
+        """ locates receivers and their corresponding monitor cores for \
+        setting router time outs
+
+        :param projections: the projections going to be read
+        :param gatherers: the gathers per ethernet chip
+        :param extra_monitors_per_chip: the extra monitor cores per chip
+        :return: list of tupels with gatherer and its extra monitor cores
+        """
+        important_gathers = set()
+
+        # iterate though projections
+        for projection in projections:
+
+            # iteration though the projections machine edges to locate chips
+            edges = self._graph_mapper.get_machine_edges(
+                projection._projection_edge)
+
+            for edge in edges:
+                placement = self._placements.get_placement_of_vertex(
+                    edge.post_vertex)
+                chip = self._machine.get_chip_at(placement.x, placement.y)
+
+                # locate extra monitor cores on the board of this chip
+                extra_monitor_cores_on_board = set()
+                for (chip_x, chip_y) in self._machine.get_chips_on_board(chip):
+                    extra_monitor_cores_on_board.add(
+                        extra_monitors_per_chip[(chip_x, chip_y)])
```

Comment:
It seems wasteful to untuple the x,y just to retuple them up again just below.

Reply:
fixed